raft/src/main/java/org/apache/kafka/snapshot/RawSnapshotWriter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.snapshot;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import org.apache.kafka.raft.OffsetAndEpoch;
+
+/**
+ * Interface for writing snapshot as a sequence of records.
+ */
+public interface RawSnapshotWriter extends Closeable {
+    /**
+     * Returns the end offset and epoch for the snapshot.
+     */
+    public OffsetAndEpoch snapshotId();
+
+    /**
+     * Returns the number of bytes for the snapshot.
+     *
+     * @throws IOException for any IO error while reading the size
+     */
+    public long sizeInBytes() throws IOException;
+
+    /**
+     * Fully appends the buffer to the snapshot.
+     *
+     * If the method returns without an exception the given buffer was fully writing the
+     * snapshot.
+     *
+     * @param buffer the buffer to append
+     * @throws IOException for any IO error during append
+     */
+    public void append(ByteBuffer buffer) throws IOException;
```

Comment:
As @hachikuji and I discussed offline, it is very like that this method will change to:

```java
public void append(MemoryRecords records) throws IOException;
```

After we implement https://issues.apache.org/jira/browse/KAFKA-10694